string data_dir
---
bool result